dora/core/common/src/main/java/alluxio/membership/WorkerServiceEntity.java

```diff
@@ -91,7 +105,8 @@ public boolean equals(Object o) {
       return false;
     }
     WorkerServiceEntity anotherO = (WorkerServiceEntity) o;
-    return mAddress.equals(anotherO.mAddress)
+    return mIdentity.equals(anotherO.mIdentity)
+        && mAddress.equals(anotherO.mAddress)
```

Comment:
Do we still need to compare the address?

Reply:
Good catch. Honestly I'm not very sure about the intent here. @lucyge2022 do you think we can remove `mAddress` from this class in favor of `mIdentity` ?

Reply:
yeah u should definitely remove this, good catch

Reply:
on a closer look, I think the worker net address should be preserved as the client needs it for creating connections. However, the netaddress should not be part of the equality check as it's only some supplementary data, not a defining factor of a `WorkerServiceEntity`.

Reply:
excluded net address from the equality check in [1931f46](https://github.com/Alluxio/alluxio/pull/18210/commits/1931f467409efec00eb9dfd23bb3a94981a6dadb)